swarm/src/lib.rs

```diff
@@ -651,7 +673,13 @@ where
                             &endpoint,
                             failed_addresses.as_ref(),
                         );
-                        if num_established.get() == 1 {
+                        // The peer is not banned, but there could be previous banned connections
+                        // if the peer was just unbanned. Check if this is the first allowed
+                        // connection.
+                        let first_allowed = established_ids
+                            .into_iter()
+                            .all(|conn_id| this.banned_peer_connections.contains(&conn_id));
+                        if first_allowed {
```

Comment:
```suggestion
                        // The peer is not banned, but there could be previous banned connections
                        // if the peer was just unbanned. Check if this is the first non-banned
                        // connection.
                        let first_non_banned = established_ids
                            .into_iter()
                            .all(|conn_id| this.banned_peer_connections.contains(&conn_id));
                        if first_non_banned {
```

Again, suggesting to call them `non-banned` and not `allowed` connections. What do you think?